src/compas_viewer/configurations/layout_config.py

```diff
@@ -0,0 +1,210 @@
+from pathlib import Path
+from typing import Dict
+from typing import List
+from typing import Literal
+from typing import TypedDict
+
+from compas_viewer import DATA
+from compas_viewer.configurations import Config
+
+
+class StatusbarConfigType(TypedDict):
+    """
+    The type template for the status bar only.
+    """
+
+    text: str
+    show_fps: bool
+
+
+class WindowConfigType(TypedDict):
+    """
+    The type template for the main window only.
+    """
+
+    about: str
+    title: str
+    width: int
+    height: int
+    fullscreen: bool
+
+
+class MenubarConfigType(TypedDict):
+    """
+    The type template for each single item of the menu.
+    """
+
+    action: Literal["action", "link"]
+    kwargs: List[str]
+
+
+class ViewportConfigType(TypedDict):
+    """
+    The type template for each single item of the viewport.
+    """
+
+    category: Literal["render"]
+    config_path: str
+
+
+class ToolbarConfigType(TypedDict):
+    """
+    The type template for the each toolbar element.
+    """
+
+    action: str
+    kwargs: dict
+
+
+class ToolbarConfig(Config):
+    """
+    The class representation for the toolbar configuration of the Layout class.
+    The toolbar configuration contains all the settings about the toolbar itself.
+    """
+
+    def __init__(self, config: Dict[str, Dict[str, ToolbarConfigType]]):
+        super().__init__(config)
+
+    @classmethod
+    def from_json(cls, filepath) -> "ToolbarConfig":
+        toolbar_config = super().from_json(filepath)
+        assert isinstance(toolbar_config, ToolbarConfig)
```

Comment:
I remember to read somewhere assert only works in tests. If runtime code, it's better to do:
```python
if not isinstance(...):
    raise TypeError(...)
```